datawave-driver-INIT-PLEASE-REVIEW-CAREFULLY.ps1

```diff
@@ -0,0 +1,271 @@
+    Write-Host "Minikube configured successfully"
+}
+
+function Preload-DockerImage([string]$image) {
+    Write-Host "Pulling $image"
+    docker pull $image
+    if ($LASTEXITCODE -eq 0) {
+        Write-Host "$image pulled successfully"
+    } else {
+        Write-Host "Failed to pull required image. Exiting."
+        exit 1
+    }
+
+    minikube image load $image
+    if ($LASTEXITCODE -eq 0) {
+        Write-Host "$image loaded into minikube successfully."
+    } else {
+        Write-Host "Failed to load $image into minikube. Exiting."
+        exit 1
+    }
+}
+
+function Configure-RepositoryCredentials {
+    if (-Not (Test-Path "${BASEDIR}\ghcr-image-pull-secret.yaml")) {
+        $username = Read-Host "Github Username"
+        $pat = Read-Host "Github Access Token" -AsSecureString
+        &"${BASEDIR}\create-image-pull-secrets.sh" $username (ConvertFrom-SecureString $pat -AsPlainText)
+    }
+    kubectl -n $NAMESPACE apply -f "${BASEDIR}\ghcr-image-pull-secret.yaml"
+}
+
+function Create-Secrets {
+    $SECRET_NAME = "certificates-secret"
+
+    if (!(kubectl -n $NAMESPACE get secret $SECRET_NAME *>&1)) {
+        Write-Host "Secret '$SECRET_NAME' does not exist. Creating secret."
+        kubectl -n $NAMESPACE create secret generic $SECRET_NAME --from-file=keystore.p12=(Join-Path -Path "${DATAWAVE_STACK}" -ChildPath "certificates\keystore.p12") --from-file=truststore.jks=(Join-Path -Path "${DATAWAVE_STACK}" -ChildPath "certificates\truststore.jks")
+    } else {
+        Write-Host "Secret '$SECRET_NAME' already exists."
+    }
+}
+
+function Ghcr-Login {
+    Write-Host "Logging in to GHCR for docker and helm"
+    if (Test-Path "${BASEDIR}\ghcr-image-pull-secret.yaml") {
+        $FILE_PATH = "${BASEDIR}\ghcr-image-pull-secret.yaml"
+
+        $ENCODED_JSON = Select-String '.dockerconfigjson:' -Path $FILE_PATH | ForEach-Object { $_.Matches } | ForEach-Object { $_.Groups[1].Value }
+        $DECODED_JSON = [System.Text.Encoding]::ASCII.GetString([System.Convert]::FromBase64String($ENCODED_JSON))
+
+        $ENCODED_AUTH = ($DECODED_JSON | ConvertFrom-Json).auths.'ghcr.io'.auth
+        $AUTH = [System.Text.Encoding]::ASCII.GetString([System.Convert]::FromBase64String($ENCODED_AUTH))
+
+        $USERNAME, $PASSWORD = $AUTH -split ':'
+
+        echo $PASSWORD | docker login ghcr.io --username $USERNAME --password-stdin
+        if ($LASTEXITCODE -eq 0) {
+            Write-Host "Docker login successful."
+        } else {
+            Write-Host "Failed to login to docker. Check credentials. Exiting."
+            exit 1
+        }
+
+        echo $PASSWORD | helm registry login ghcr.io --username $USERNAME --password-stdin
+        if ($LASTEXITCODE -eq 0) {
+            Write-Host "Helm login successful."
+        } else {
+            Write-Host "Failed to login to helm. Check credentials. Exiting."
+            exit 1
+        }
+    }
+}
```

Comment:
This whole function doesn't seem to work, I outputed the info at each step and it just gave nothing.